Dockerize voice building for Android

lumpidu opened the pull request:
This creates the voice Álfur completey from scratch, in building it for 4 Android platforms by adding a voice loader driver layer to it. The result are 4 `.so` shared objects + accompanying header file which are extraced from the built container `lvl_is:latest` to the local directory `voice_deliverables/`. There is a meta-data file `voice_info.json` that can be parsed by a program to get hold of all platform files.

lumpidu committed:
Dockerize voice building for Android

This creates the voice Álfur completey from scratch, in building it
for 4 Android platforms by adding a voice loader driver layer to it.
The result are 4 .so shared objects + accompanying header file which
are extraced from the built container lvl_is:latest to the local directory
voice_deliverables.

Signed-off-by: Daniel Schnell <dschnell@grammatek.com>

G-Thor commented on docker/Dockerfile.festvox:
```diff
@@ -0,0 +1,112 @@
+
+# Build Festvox
+WORKDIR /usr/local/src/festvox
+RUN cp /usr/local/src/ext/ehmm_patch/do_ehmm ./src/ehmm/scripts/
```
The ehmm patch is no longer necessary if we use the modified phoneset

lumpidu replied:
ok, I'll try to build without the patch

G-Thor commented on docker/Dockerfile.festvox:
```diff
@@ -0,0 +1,112 @@
+RUN ./configure && make
+
+# DS: ext/ehmm_patch, for clustergen voices
+RUN git clone https://github.com/lumpidu/unit-selection-festival.git && cp -r unit-selection-festival/ext /usr/local/src/ext
```
Since we don't need the ehmm patch, this is probably unnecessary as well.

lumpidu replied:
right, will change it accordingly

G-Thor reviewed (approved):
This seems fine to me, although as I mention the ehmm patch is probably no longer necessary if we work around the issue by modifying the phoneset.

G-Thor commented on build_docker_lvl_is.sh:
```diff
@@ -0,0 +1,10 @@
+#! /bin/bash
+
+set -euo pipefail
+docker build -f docker/Dockerfile.festvox -t harbour.grammatek.com/sim/festvox-build-tools:latest .
+docker build -f docker/Dockerfile.lvl_is --progress=plain -t harbour.grammatek.com/sim/lvl_is:latest .
```
Will Grammatek be hosting these images?

lumpidu replied:
this is our docker registry, which we have made publicly readable for the sim project. We are hosting these images there so one can just pull them without the need to build these locally

lumpidu committed:
Remove references to ehmm patch, which is not necessary with the new phonemeset

Signed-off-by: Daniel Schnell <dschnell@grammatek.com>

lumpidu committed:
voice_driver: don't use flite_phones_to_speech() for generating voice audio from phonemes

flite_phones_to_speech() needs a file to write the audio into. We used
the approach of transferring audio bytes to the caller by using a temporary
file that was given to flite_phones_to_speech() and then reading the
file contents of this file to the user provided buffer.

This was a cludge and doesn't work on Android, because there the temporary
directory is not well definied.
Use a direct approach for writing the voice audio to the user buffer instead

Signed-off-by: Daniel Schnell <dschnell@grammatek.com>

lumpidu committed:
Automatically create voice_info.json inside voice_deliverables/ directory

This file contains the correct MD5 sums of the library object for each
build architecture.

Also update Android API version to the same as the MindSDKVersion
in Símarómur.

Signed-off-by: Daniel Schnell <dschnell@grammatek.com>

lumpidu committed:
voice_driver.[ch]: add alignment for struct voice_drv_priv_t/voice_drv_info_t

Sometimes a wrong alignment of these can lead to bus errors in the
calling linker step when initializing the voice.


Signed-off-by: Daniel Schnell <dschnell@grammatek.com>
